### DISCUSSION TITLE
CLI plan and CLI tools

### DISCUSSION THREAD

**Discussion by alshdavid** [alshdavid/TypeScriptBC#77] [Ideas]
What will the CLI look like and what tools will be included?

- Compiler (obviously)
- Test runner (and test framework/assertions/etc)
- Linter
- Documentation generation (thanks Go for setting the bar 😄)
- Task runner (think npm scripts)
- Module management

**Comment by SuperSonicHub1** [comment]
Let's have all of these tools accessible from a single binary, like Deno.

**Comment by alshdavid** [comment]
Definitely agree with the single binary

**Comment by hugojosefson** [comment]
Oh, and please a formatter. That's one thing I love about Deno and Rust!
